gremlin-go/driver/driverRemoteConnection.go

```diff
@@ -100,6 +129,41 @@ func (driver *DriverRemoteConnection) submitBytecode(bytecode *bytecode) (Result
 	return driver.client.submitBytecode(bytecode)
 }
 
+// IsSessionBound returns True if a DriverRemoteConnection is a Session
+func (driver *DriverRemoteConnection) IsSessionBound() bool {
+	return driver.client.session != ""
+}
+
+// CreateSession generates a new Session. sessionId stores the optional UUID param. It can be used to create a session with a specific UUID.
+func (driver *DriverRemoteConnection) CreateSession(sessionId ...string) (*DriverRemoteConnection, error) {
+	driver.client.logHandler.logger.Log(Info, "creating session based connection")
+	if driver.IsSessionBound() {
+		return nil, errors.New("connection is already bound to a session - child sessions are not allowed")
+	}
+	var sessionHandler func(settings *DriverRemoteConnectionSettings)
+	if len(sessionId) > 1 {
+		return nil, errors.New("more than one session ID specified. Cannot create Session with multiple UUIDs")
+	} else if len(sessionId) == 1 {
+		sessionHandler = func(settings *DriverRemoteConnectionSettings) {
+			settings.Session = sessionId[0]
+		}
+	} else {
+		sessionHandler = func(settings *DriverRemoteConnectionSettings) {
+			settings.Session = uuid.New().String()
+		}
+	}
+	connection, err := NewDriverRemoteConnection(driver.client.url, sessionHandler)
+	if err != nil {
+		return nil, err
+	}
+	driver.spawnedSessions = append(driver.spawnedSessions, connection)
+	return connection, nil
+}
```

Comment:
This entire portion is a bit confusing to me.

It doesn't seem like the session is actually being used by any clients, whether it be being created with one, or having CreateSession be invoked.

The distinction between adding a session to an existing DRC and creating a new one with a session are different use cases but it's not very obvious what is for what, as you can do both. Creating a session from an existing DRC is also confusing because it creates an instance of the same type that.